Sources/Basic/SortedArray.swift

```diff
@@ -59,14 +59,55 @@ public struct SortedArray<Element>: CustomStringConvertible {
     
     /// Insert the given sequence, maintaining the sort order.
     public mutating func insert<S: Sequence>(contentsOf newElements: S) where S.Iterator.Element == Element {
+        var newElements = newElements.sorted(by: areInIncreasingOrder)
+        guard !newElements.isEmpty else {
+            return
+        }
+        guard !elements.isEmpty else {
+            elements = newElements
+            return
+        }
+
+        var lhsIndex = elements.endIndex - 1
+        var rhsIndex = newElements.endIndex - 1
+
+        elements.reserveCapacity(elements.count + newElements.count)
+
+        // TODO: If SortedArray moves to stdlib an _ArrayBuffer can be used
+        // instead, this can then be removed as _ArrayBuffer can be resized
+        // without requiring instantiated elements.
         elements.append(contentsOf: newElements)
-        elements.sort(by: areInIncreasingOrder)
+
+        var lhs = elements[lhsIndex], rhs = newElements[rhsIndex]
+
+        // Equivalent to a merge sort, "pop" and append the max elemeent of
+        // each array until either array is empty
+        for index in elements.indices.reversed() {
+            if areInIncreasingOrder(lhs, rhs) {
+                elements[index] = rhs
+                rhsIndex -= 1
+                guard rhsIndex >= 0 else { break }
```

Comment:
`0` should be `startIndex` here I think, and there should probably be test coverage that shows an error without this.